sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -89,22 +89,22 @@ class KotlinSensor(
             fileSystem.predicates().hasType(InputFile.Type.MAIN)
         )
 
-        val filesToAnalyze = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
+        val filePartition: Pair<Iterable<InputFile>, Iterable<InputFile>> = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
             if (canSkipUnchangedFiles(sensorContext)) {
                 LOG.debug("The Kotlin analyzer is running in a context where it can skip unchanged files.")
                 var totalFiles = 0
-                mainFiles
-                    .filter {
-                        totalFiles++
-                        it.status() != InputFile.Status.SAME
-                    }.also {
-                        LOG.info("Only analyzing ${it.size} changed Kotlin files out of ${totalFiles}.")
-                    }
+                val partition = mainFiles.partition {
+                    totalFiles++
+                    it.status() == InputFile.Status.SAME && reuseCPDTokens(it, sensorContext)
+                }
+                LOG.info("Only analyzing ${partition.second.size} changed Kotlin files out of ${totalFiles}.")
+                partition
             } else {
                 LOG.debug("The Kotlin analyzer is running in a context where unchanged files cannot be skipped.")
-                mainFiles
+                Pair(emptyList(), mainFiles)
             }
         }
+        val filesToAnalyze = filePartition.second
```

Comment:
For Pairs, Kotlin supports destructuring. So instead of explicitly calling `.first` or `.second`, use e.g.
```
val (_, filesToAnalyze) = filePartition
```
Even better, you don't need the intermediary val `filePartition` at all. In fact, do we even need the partition at all? I don't see where the files that are not being analyzed are being used.